ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java

```diff
@@ -114,7 +114,7 @@ private void createOrResumeXidTransaction(final XidTransactionID xidTransactionI
         } else {
             // begin a new tx and add it to the tx repository
             // TODO: Fix the tx timeout to accept a value from the client (WFLY-2789)
-            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.MAX_VALUE);
+            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.parseInt(System.getProperty("remote-tx-timeout", "31536000")));
```

Comment:
@ctomc Please see David's comments in the BZ. This is where this solution is coming from.

@rsvoboda We really can't add new management attributes in EAP 6 CPs, so if this will be fixed in a CP then a workaround solution not based on system properties is needed there, and this PR then becomes a forward port to ensure there is no regression. Some other solution then is a separate issue. Of course, rejecting a workaround solution and just not fixing it yet is a valid choice; I have no strong opinion on that.

If we are going with a system property based solution, then we need a different property name as "remote-tx-timeout" is too general. If we have a property naming pattern that emphasizes the fact that such properties are workarounds that may be replaced by a better solution later, let's use it here.